python/ray/serve/tests/test_http_headers.py

```diff
@@ -22,17 +23,28 @@ def __call__(self):
     assert resp.status_code == 200
     assert RAY_SERVE_REQUEST_ID_HEADER in resp.headers
     assert resp.text == resp.headers[RAY_SERVE_REQUEST_ID_HEADER]
+    assert resp.text == resp.headers["x-request-id"]
+
+    def is_valid_uuid(num: str):
+        try:
+            uuid.UUID(num, version=4)
+            return True
+        except ValueError:
+            return False
+
+    assert is_valid_uuid(resp.text)
 
 
 class TestUserProvidedRequestIDHeader:
     def verify_result(self):
-        resp = requests.get(
-            "http://localhost:8000", headers={RAY_SERVE_REQUEST_ID_HEADER: "123-234"}
-        )
-        assert resp.status_code == 200
-        assert resp.json() == 1
-        assert RAY_SERVE_REQUEST_ID_HEADER in resp.headers
-        assert resp.headers[RAY_SERVE_REQUEST_ID_HEADER] == "123-234"
+
+        for header_attr in [RAY_SERVE_REQUEST_ID_HEADER, "X-Request-ID"]:
```

Comment:
Would be nice if we can add cases for when both are set at the same time with different values :)